docs/common/types.ts

```diff
@@ -0,0 +1,31 @@
+export type PageMetadata = {
+  title: string;
+  sourceCodeUrl?: string;
+  maxHeadingDepth?: number;
+  hideTOC?: boolean;
+  headings?: {
+    title?: string;
+    level?: number;
+    type?: string;
+    _processed?: boolean; // internal HeadingManager property
+  }[];
+};
+
+/**
+ * Utility type. Extracts `T` type from `T[]` array.
+ */
+export type Single<T> = T extends (infer U)[] ? U : never;
```

Comment:
```suggestion
export type ElementType<T extends any[]> = T extends (infer U)[] ? U : never;
```
I believe `Single` has too broad meaning 🤔 
Additionally, maybe we want to be more specific about generic argument? 🤔